Core/include/Acts/EventData/MultiComponentTrackParameters.hpp

```diff
@@ -26,15 +27,12 @@ namespace Acts {
 /// @note This class holds shared ownership on its reference surface.
 /// @note The accessors for parameters, covariance, position, etc.
 /// are the weighted means of the components.
-/// @note If all covariances are zero, the accessor for the total
-/// covariance does return std::nullopt;
 /// TODO Add constructor from range and projector maybe?
 template <typename charge_t>
 class MultiComponentBoundTrackParameters {
   using SingleParameters = GenericBoundTrackParameters<charge_t>;
 
-  std::vector<std::tuple<double, BoundVector, std::optional<BoundSquareMatrix>>>
-      m_components;
+  std::vector<std::tuple<double, BoundVector, BoundSquareMatrix>> m_components;
```

Comment:
Not directly related to the change, but I think usually structs with names are prefferable over tuples with many entries.